core/app/services/spree/promotion_batches/promo_codes_exporter.rb

```diff
@@ -0,0 +1,25 @@
+module Spree
+  module PromotionBatches
+    class PromoCodesExporter
+      def initialize(params)
+        @promotion_batch = find_promotion_batch(params[:id])
+      end
+
+      def call
+        Spree::Core::Converters::CSV.to_csv(promo_codes)
+      end
+
+      private
+
+      def find_promotion_batch(id)
+        Spree::PromotionBatch.find(id)
+      end
```

Comment:
Is this doing anything? `Spree::PromotionBatch.find(id)` itself is very clear.

Reply:
It is encapsulating the logic of finding the record. Having these encapsulated is a good practice (S in SOLID).
Also, someday we might be implementing the **repository pattern** (or any other querying mechanism), and such seemingly unnecessary steps would be helpful then.